File: Delphi/uMainForm.pas
unit uMainForm;

interface

uses
  Winapi.Windows, Winapi.Messages, System.SysUtils, System.Variants, System.Classes, Vcl.Graphics,
  Vcl.Controls, Vcl.Forms, Vcl.Dialogs, cxClasses, dxReport, cxGraphics,
  cxLookAndFeels, cxLookAndFeelPainters, Vcl.Menus,  cxButtons,
  cxControls, cxStyles, cxCustomData, cxFilter, cxData,
  cxDataStorage, cxEdit, cxNavigator, dxDateRanges, dxScrollbarAnnotations,
  Data.DB, cxDBData, cxGridLevel, cxGridCustomView, cxGridCustomTableView,
  cxGridTableView, cxGridDBTableView, cxGrid, FireDAC.Comp.DataSet,
  FireDAC.Comp.Client,
  Vcl.StdCtrls, dxmdaset,
  dxLayoutControlAdapters, dxLayoutContainer, dxLayoutControl,
  dxBackend.Utils.WebBrowserForm, dxDashboard.Control, dxAI,
  dxAI.Commands.SmartPaste, cxTextEdit, cxMaskEdit, dxBackend,
  dxBackend.ConnectionString.SQL, Vcl.Grids, Vcl.DBGrids,
  dxLayoutcxEditAdapters, dxCore, FireDAC.Stan.Intf, FireDAC.Stan.Option,
  FireDAC.Stan.Error, FireDAC.UI.Intf, FireDAC.Phys.Intf, FireDAC.Stan.Def,
  FireDAC.Stan.Pool, FireDAC.Stan.Async, FireDAC.Phys, FireDAC.Phys.SQLite,
  FireDAC.Phys.SQLiteDef, FireDAC.Stan.ExprFuncs,
  FireDAC.Phys.SQLiteWrapper.Stat, FireDAC.VCLUI.Wait, FireDAC.Stan.Param,
  FireDAC.DatS, FireDAC.DApt.Intf, FireDAC.DApt, cxContainer, cxDropDownEdit,
  cxLookupEdit, cxDBLookupEdit, cxDBLookupComboBox, dxSkinsForm;

type
  TMainForm = class(TForm)
    btnDesign: TcxButton;
    dxLayoutControl1Group_Root: TdxLayoutGroup;
    dxLayoutControl1: TdxLayoutControl;
    liBtnShowDesigner: TdxLayoutItem;
    dxDashboardControl1: TdxDashboardControl;
    liDashboardControl: TdxLayoutItem;
    lgButtons: TdxLayoutGroup;
    btnView: TcxButton;
    liBtnDelete: TdxLayoutItem;
    dxBackendDataConnectionManager1: TdxBackendDataConnectionManager;
    NWindConnectionString: TdxBackendDatabaseSQLConnection;
    dsNWind: TDataSource;
    dxSkinController1: TdxSkinController;
    FDConnection1: TFDConnection;
    FDNWind: TFDQuery;
    cbCountries: TcxLookupComboBox;
    licbCountries: TdxLayoutItem;
    procedure FormCreate(Sender: TObject);
    procedure btnDesignClick(Sender: TObject);
    procedure cbCountriesPropertiesEditValueChanged(Sender: TObject);
    procedure btnViewClick(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  MainForm: TMainForm;

implementation

{$R *.dfm}

procedure TMainForm.FormCreate(Sender: TObject);
begin
  FDNWind.Active := True;
end;

procedure TMainForm.btnDesignClick(Sender: TObject);
begin
  dxDashboardControl1.ShowDesigner;
end;

procedure TMainForm.btnViewClick(Sender: TObject);
begin
   dxDashboardControl1.ShowViewer;
end;

procedure TMainForm.cbCountriesPropertiesEditValueChanged(Sender: TObject);
begin
  dxDashboardControl1.Parameters['CountryDashboardParameter'].Value := cbCountries.EditValue;
  dxDashboardControl1.Dashboard.ApplyParametersToState;
end;

end.

File: CPB/uMainForm.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "uMainForm.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxButtons"
#pragma link "cxClasses"
#pragma link "cxContainer"
#pragma link "cxControls"
#pragma link "cxDBLookupComboBox"
#pragma link "cxDBLookupEdit"
#pragma link "cxDropDownEdit"
#pragma link "cxEdit"
#pragma link "cxGraphics"
#pragma link "cxLookAndFeelPainters"
#pragma link "cxLookAndFeels"
#pragma link "cxLookupEdit"
#pragma link "cxMaskEdit"
#pragma link "cxTextEdit"
#pragma link "dxBackend"

#if defined(_WIN64)
  #pragma link "dxBackend.ConnectionString.SQL.O"
#else
  #pragma link "dxBackend.ConnectionString.SQL.OBJ"
#endif

#pragma link "dxBackend.Utils.WebBrowserForm"
#pragma link "dxCore"
#pragma link "dxDashboard.Control"
#pragma link "dxLayoutContainer"
#pragma link "dxLayoutControl"
#pragma link "dxLayoutControlAdapters"
#pragma link "dxLayoutcxEditAdapters"
#pragma link "dxSkinsForm"
#pragma resource "*.dfm"
TMainForm *MainForm;
//---------------------------------------------------------------------------
__fastcall TMainForm::TMainForm(TComponent* Owner)
	: TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::FormCreate(TObject *Sender)
{
	FDNWind->Active = true;
}
//---------------------------------------------------------------------------

void __fastcall TMainForm::btnDesignClick(TObject *Sender)
{
	dxDashboardControl1->ShowDesigner();
}
//---------------------------------------------------------------------------

void __fastcall TMainForm::btnViewClick(TObject *Sender)
{
	dxDashboardControl1->ShowViewer();
}
//---------------------------------------------------------------------------

void __fastcall TMainForm::cbCountriesPropertiesEditValueChanged(TObject *Sender)

{
	dxDashboardControl1
		->Parameters->ParamByName[L"CountryDashboardParameter"]
		->Value = cbCountries->EditValue;

	dxDashboardControl1->Dashboard->ApplyParametersToState();
}
//---------------------------------------------------------------------------

